tracer/src/Datadog.Trace/Tracer.cs

```diff
@@ -76,8 +76,8 @@ public Tracer(TracerSettings settings)
         /// Note that this API does NOT replace the global Tracer instance.
         /// The <see cref="TracerManager"/> created will be scoped specifically to this instance.
         /// </summary>
-        internal Tracer(TracerSettings settings, IAgentWriter agentWriter, ITraceSampler sampler, IScopeManager scopeManager, IDogStatsd statsd, ITelemetryController telemetry = null, IDiscoveryService discoveryService = null)
-            : this(TracerManagerFactory.Instance.CreateTracerManager(settings?.Build(), agentWriter, sampler, scopeManager, statsd, runtimeMetrics: null, logSubmissionManager: null, telemetry: telemetry ?? NullTelemetryController.Instance, discoveryService ?? NullDiscoveryService.Instance, dataStreamsManager: null))
+        internal Tracer(TracerSettings settings, IAgentWriter agentWriter, ITraceSampler sampler, ISpanSampler spanSampler, IScopeManager scopeManager, IDogStatsd statsd, ITelemetryController telemetry = null, IDiscoveryService discoveryService = null)
```

Comment:
I wonder if `spanSampler` and all the parameters after it should be optional like the last two. But that may require reordering since some of them cannot be optional. Just something to consider, level of care low.

Reply:
Sounds good, thanks.